R/load_data.R

```diff
@@ -254,7 +254,15 @@ ReadMPX_Seurat <- function(
   feature_meta_data <-
     names(hd5_object[["var"]]) %>%
     lapply(function(nm) {
-      col <- tibble(!!sym(nm) := hd5_object[["var"]][[nm]]$read())
+      # If reading function is missing, don't read the column
+      # This is likely due to the column being empty
+      hd5_read_func <- hd5_object[["var"]][[nm]]$read
+      if (is.null(hd5_read_func)) {
+        warn(glue("Column '{nm}' in var is empty. Skipping."))
+        return(NULL)
+      }
+
+      col <- tibble(!!sym(nm) := hd5_read_func())
```

Comment:
This likely fails when `hd5_object[["var"]][[nm]]` does not have a `$read()` property, in which case it's NULL. But are we sure it's always NULL?

I think a more appropriate solution is to wrap the expression with try. For example:

````
col <- try({tibble(!!sym(nm) := hd5_read_func())}, silent = TRUE)
if (inherits(col, "try-error") {
  warn(glue("Column '{nm}' in var is empty. Skipping."))
  return(NULL)
}
return(col)
````

Reply:
Good suggestion, I agree